cmake/Buildmatlab-whole-body-simulator.cmake

```diff
@@ -0,0 +1,34 @@
+# Copyright (C) Fondazione Istituto Italiano di Tecnologia
+# CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+include(YCMEPHelper)
+include(FindOrBuildPackage)
+
+find_or_build_package(YARP QUIET)
+find_or_build_package(yarp-matlab-bindings QUIET)
+find_or_build_package(WBToolbox QUIET)
+find_or_build_package(iDynTree QUIET)
+find_or_build_package(qpOASES QUIET)
+find_or_build_package(icub-models QUIET)
+find_or_build_package(whole-body-controllers QUIET)
+
+set(matlab-whole-body-simulator_DEPENDS "")
+list(APPEND matlab-whole-body-simulator_DEPENDS YARP)
+list(APPEND matlab-whole-body-simulator_DEPENDS yarp-matlab-bindings)
+list(APPEND matlab-whole-body-simulator_DEPENDS WBToolbox)
+list(APPEND matlab-whole-body-simulator_DEPENDS iDynTree)
+list(APPEND matlab-whole-body-simulator_DEPENDS qpOASES)
+list(APPEND matlab-whole-body-simulator_DEPENDS icub-models)
+list(APPEND matlab-whole-body-simulator_DEPENDS whole-body-controllers)
```

Comment:
This dependency can create a problem if whole-body-controllers starts depending on matlab-whole-body-simulator (see https://github.com/robotology/whole-body-controllers/issues/95#issuecomment-822518860). If it is just for running some examples and it is not a core dependency of the library that it is installed, I think we can drop this dependency (as in any case it is not required at install time)

Reply:
We use the method `wbc.skew`. But in any case, for the YOGA++ example model in `matlab-whole-body-simulator` we want to use the torque controller from whole-body-controllers.

One solution would be to have a repo with a library of controllers (whole-body-controllers), a repo with a library of simulators (matlab-whole-body-simulator), and a repo of model examples, test models, etc, which would use the libraries from the first two repos.

But it should be possible for two projects to depend on each other. No?

Reply:
Removed the dependency inhttps://github.com/robotology/robotology-superbuild/pull/689/commits/7bb486bd34ba0f5d518abf86c49c58f81aa134f1.

Reply:
> But it should be possible for two projects to depend on each other. No?

In theory for languages such as MATLAB yes, as long as both are on the PATH. However, for how the superbuild works, it would require all the dependencies of a given project to be installed before installing a given project, so a circular dependency could create problems. In any case, if the dependency is just a runtime dependency, you can still have it, as long as it is not listed in the superbuild.